### ISSUE TITLE
A file named "C:\Program" prevents WSL from starting

### ISSUE THREAD

**Issue by dantmnf** [microsoft/WSL#10637]
### Windows Version

Microsoft Windows [Version 10.0.22621.2428]

### WSL Version

1.2.5.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.90.1

### Distro Version

N/A

### Repro Steps

```console
C:\>wsl echo hello
hello

C:\>wsl --shutdown

C:\>echo a > Program

C:\>wsl echo hello
%1 is not a valid Win32 application.
Error code: Wsl/Service/CreateInstance/CreateVm/0x800700c1

C:\>del Program

C:\>wsl echo hello
hello

```

### Expected Behavior

WSL should start regardless the existence of the `C:\Program` file.

### Actual Behavior

As long as the `C:\Program` file exists, WSL will not start.

**Comment by pmartincic** [comment]
Are you sure this has to do with WSL? Or with the service control manager?

**Comment by dantmnf** [comment]
The `WslService`, `vmcompute`, `HNS` services are all running. Any extra service will start during the process?

**Comment by dantmnf** [comment]
[WslLogs-2023-10-22_22-59-28.zip](https://github.com/microsoft/WSL/files/13063730/WslLogs-2023-10-22_22-59-28.zip)

**Comment by OneBlue** [comment]
Fixed in [2.0.6](https://github.com/microsoft/WSL/releases/tag/2.0.6)
